streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceivedBlockHandler.scala

```diff
@@ -151,13 +167,21 @@ private[streaming] class WriteAheadLogBasedBlockHandler(
    */
   def storeBlock(blockId: StreamBlockId, block: ReceivedBlock): ReceivedBlockStoreResult = {
 
+    var numRecords = None: Option[Long]
     // Serialize the block so that it can be inserted into both
     val serializedBlock = block match {
       case ArrayBufferBlock(arrayBuffer) =>
-        blockManager.dataSerialize(blockId, arrayBuffer.iterator)
+        val countIterator = new CountingIterator(arrayBuffer.iterator)
```

Comment:
Ok. Will change back to as it was earlier. But if we do not go through CountingIterator we may not know if Block is fullyConsumed or not. But we are not using isFullyConsumed presently , so this should be fine.